base/commands/viridian/common.go

```diff
@@ -53,3 +66,39 @@ func getAPI(ec plug.ExecContext) (*viridian.API, error) {
 	}
 	return viridian.NewAPI(string(token)), nil
 }
+
+func waitClusterState(ctx context.Context, ec plug.ExecContext, api *viridian.API, clusterIDOrName, state string) error {
+	for {
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+		cs, err := api.ListClusters(ctx)
```

Comment:
Is there a reason we are using the `list` endpoint instead of `get`?

Reply:
That would do 2 HTTP requests instead of 1.

Reply:
Ah, I see we are using cluster id or name.